addons/knowledge/controllers/main.py

```diff
@@ -0,0 +1,311 @@
+            ("user_id", "=", request.env.user.id), ('article_id.active', '=', True)
+        ])
+        return request.env['ir.qweb']._render('knowledge.knowledge_article_tree_favorites', {
+            'favorites': favorite_articles,
+            "active_article": request.env['knowledge.article'].browse(active_article_id),
+        })
+
+    # ------------------------
+    # Article permission panel
+    # ------------------------
+
+    @http.route('/knowledge/get_article_permission_panel_data', type='json', auth='user')
+    def get_article_permission_panel_data(self, article_id):
+        """
+        Returns a dictionnary containing all values required to render the permission panel.
+        :param article_id: (int) article id
+        """
+        article = self._fetch_article(article_id)
+        if not article:
+            return werkzeug.exceptions.Forbidden()
+        is_sync = not article.is_desynchronized
+        # Get member permission info
+        members_values = []
+        members_permission = article._get_article_member_permissions(additional_fields={
+            'res.partner': [
+                ('name', 'partner_name'),
+                ('email', 'partner_email'),
+                ('partner_share', 'partner_share'),
+            ],
+            'knowledge.article': [('name', 'based_on_name')],
+        })[article.id]
+
+        for partner_id, member in members_permission.items():
+            # empty member added by '_get_article_member_permissions', don't show it in the panel
+            if not member['member_id']:
+                continue
+
+            # if share partner and permission = none, don't show it in the permission panel.
+            if member['permission'] == 'none' and member['partner_share']:
+                continue
+
+            # if article is desyncronized, don't show members based on parent articles.
+            if not is_sync and member['based_on']:
+                continue
+
+            member_values = {
+                'id': member['member_id'],
+                'partner_id': partner_id,
+                'partner_name': member['partner_name'],
+                'partner_email': member['partner_email'],
+                'permission': member['permission'],
+                'based_on': member['based_on_name'],
+                'based_on_id': member['based_on'],
+                'partner_share': member['partner_share'],
+                'is_unique_writer': member['permission'] == "write" and article.inherited_permission != "write" and not any(
+                    other_member['permission'] == 'write'
+                    for partner_id, other_member in members_permission.items()
+                    if other_member['member_id'] != member['member_id']
+                ),
+            }
+            members_values.append(member_values)
+
+        internal_permission_field = request.env['knowledge.article']._fields['internal_permission']
```

Comment:
would `fields_get(allfields=['internal_permission', 'permission'])` work?

Reply:
Completely removed. It was actually doing what read does.